[Mechanics - Character] Adding Laser Rifle into the game and several bug fixed

Alasses opened the pull request:
- Add laser rifle into the game
- Modified weapon system to fit networking structures
- Fixed bug of gravity in-game
- Fixed bug of viewing inside the model
- Removed the test folder
- Fixed the player rotation issue, separate the player object and rotation into head and body

Alasses committed:
Laser Rifle v0.1

- Created basic Laser Rifle
- Modified the structure of weapon system
- Modified the ammo classes

Alasses committed:
Laser Rifle v0.2

- Fixed the UI loading issues

Alasses committed:
Laser Rifle v0.3

- Modified the weapon and ammo system to fit the networking requirements

Alasses committed:
Fail build

Alasses committed:
Revert to v0.2

Everything fucked up, trying to fix them

Alasses committed:
Revert to v0.3

Everything fucked up, trying to fix them

Alasses committed:
Revert to v0.3

Alasses committed:
Revert to v0.2

Alasses committed:
Revert to v0.3

Alasses committed:
Laser Rifle v0.4

- Successfully reverted to old commit and fixed problem
- Fixed jump bug

Alasses committed:
Laser Rifle v1.0

- Fixed network issues of the weapon system
- Created a basic networking weapon structure

JasonStallkamp commented on Local War/Assets/Resources/Script/Buff/Buff.cs:
```diff
@@ -4,7 +4,8 @@
 public class Buff : MonoBehaviour {
 
-    protected string name;
+    protected string bname;
```
Why did we change this name

Alasses replied:
The unity generate an error that says the "name" actually override a built-in variable. So I have to change the name into bname, I will double-check if it will still generate the error.

JasonStallkamp replied:
OK

JasonStallkamp commented:
A build for the game should not be included in the source control. The entire  Local War/Test/* should be taken out.

JasonStallkamp commented on Local War/Assets/Resources/Script/Character/MovementSys.cs:
```diff
@@ -66,10 +66,11 @@ private void move()
         /*****Move the player*****/
         controller.Move(moveDirection * Time.deltaTime);    //move the character based on the gravitational force.
         if (!controller.isGrounded)
-        {
             jumpDirection.y -= gravity * Time.deltaTime;
-        }
+
         controller.Move(jumpDirection * Time.deltaTime);
+        if(controller.isGrounded)
+            jumpDirection.y = 0;
```
In the future, this really should be done by character systems.  It was assigned to someone else on trello

Alasses replied:
Do you mean the isGround function? Someone is already working on our own version of the function.

JasonStallkamp replied:
yeah. if we merge this, they are probably going to have a merge conflict. it will only be one or two lines, but it's okay for now

JasonStallkamp commented:
You did not mention the laser pistol. Does it work? should it be in this commit

Alasses committed:
Fixed the laser pistol

- Update the laser pistol to make it work for current system.

Alasses committed:
Update .gitignore

Alasses committed:
Remove the test build

- Rolled back the git ignore file
- Removed the test folder

Alasses committed:
Fixed the body rotation

- Separate the player into head and body
- Separate the rotation into head rotation and body rotation

Alasses committed:
Update the scene

Nathan127 commented:
I tested your branch and I wasn't able to play test it. It has too many errors and the pull request is already a mess. (Not mentioning changes, touching my .gitignore, adding an entire test folder, touching other people's code [the gravity system]) so I'm going to close this pull request. Feel free to make a new one after this all gets sorted out.
